src/components/app/details/triggerView/TriggerView.tsx

```diff
@@ -868,8 +868,44 @@ class TriggerView extends Component<TriggerViewProps, TriggerViewState> {
     renderCIMaterial = () => {
         if ((this.state.ciNodeId && this.state.showCIModal) || this.state.showMaterialRegexModal) {
             let nd: NodeAttr
+            const configuredMaterialList = new Map()
             for (let i = 0; i < this.state.workflows.length; i++) {
                 nd = this.state.workflows[i].nodes.find((node) => +node.id == this.state.ciNodeId && node.type === 'CI')
+                configuredMaterialList[this.state.workflows[i].name] = new Set()
+
+                if (nd?.[this.state.materialType].length > 0) {
+                    nd?.[this.state.materialType].map((node, _) =>
```

Comment:
use for..of or forEach as we just want to iterate

Reply:
Also, you don't need to mark optional again.

Reply:
done